### ISSUE TITLE
Bug about record in C# 9 while invoking GenerateSchema

### ISSUE THREAD

**Issue by newbe36524** [AdrianStrugala/AvroConvert#28]
```cs
    class Program
    {
        static void Main(string[] args)
        {
            string schema = AvroConvert.GenerateSchema(typeof(TestClass));
            Console.WriteLine(schema);
        }
    }
    
    public record TestClass
    {
        public string Name { get; set; }
    }
```

An exception thrown:

```bash
Unhandled exception. System.Runtime.Serialization.SerializationException: Could not find any matching known type for 'System.Type'.
   at SolTechnology.Avro.BuildSchema.ReflectionSchemaBuilder.BuildKnownTypeSchema(Type type, Dictionary`2 schemas, UInt32 currentDepth)
   at SolTechnology.Avro.BuildSchema.ReflectionSchemaBuilder.CreateNotNullableSchema(Type type, Dictionary`2 schemas, UInt32 currentDepth)
   at SolTechnology.Avro.BuildSchema.ReflectionSchemaBuilder.CreateSchema(Boolean forceNullable, Type type, Dictionary`2 schemas, UInt32 currentDepth, Type prioritizedType)
   at SolTechnology.Avro.BuildSchema.ReflectionSchemaBuilder.AddRecordFields(IEnumerable`1 members, Dictionary`2 schemas, UInt32 currentDepth, RecordSchema record)
   at SolTechnology.Avro.BuildSchema.ReflectionSchemaBuilder.BuildRecordTypeSchema(Type type, Dictionary`2 schemas, UInt32 currentDepth)
   at SolTechnology.Avro.BuildSchema.ReflectionSchemaBuilder.BuildComplexTypeSchema(Type type, Dictionary`2 schemas, UInt32 currentDepth)
   at SolTechnology.Avro.BuildSchema.ReflectionSchemaBuilder.CreateNotNullableSchema(Type type, Dictionary`2 schemas, UInt32 currentDepth)
   at SolTechnology.Avro.BuildSchema.ReflectionSchemaBuilder.CreateSchema(Boolean forceNullable, Type type, Dictionary`2 schemas, UInt32 currentDepth, Type prioritizedType)
   at SolTechnology.Avro.BuildSchema.ReflectionSchemaBuilder.BuildSchema(Type type)
   at SolTechnology.Avro.AvroConvert.GenerateSchema(Type type)
   at AvroSchemaTool.Program.Main(String[] args) in D:\Repo\AvroSchemaTool\Program.cs:line 10

```

**Comment by AdrianStrugala** [comment]
Hello,

Thanks for reporting the problem. I'm starting the investigation.

I'll keep you updated

**Comment by AdrianStrugala** [comment]
Hi,

The fix is ready, C#9 records are now fully supported.
You can find it in **v.2.7.1** of the package.

Could you confirm if the problem is solved?

Regards,
Adrian

**Comment by newbe36524** [comment]
Hi Adrian, it works! Many thanks for you nice job!
